src/input/mod.rs

```diff
@@ -235,128 +235,148 @@ impl Input {
+        input.delta.keys_hit.iter().filter(|&&k| k == self.pos || k == self.neg).count() > 0
+    }
+}
+
+impl Hit for axis::Raw {
+    fn hit(&self, input: &Input) -> bool {
+        input.delta.axes_raw.iter().filter(|&&(id, _)| id == self.id).count() > 0
+    }
+}
+
+pub trait HitCount {
+    type Output;
+    fn hit_count(&self, input: &Input) -> Self::Output;
+}
+
+impl HitCount for Button {
+    type Output = u8;
+    fn hit_count(&self, input: &Input) -> Self::Output {
         use std::u8::MAX;
-        let pressed = match button {
-            Button::Key(button) => self.state.keys_pressed.contains(&button),
-            Button::Mouse(button) => self.state.mouse_pressed.contains(&button),
-        };
-        let hits = match button {
-            Button::Key(button) => self.delta
+        match *self {
+            Button::Key(button) => input.delta
                 .keys_hit
                 .iter()
                 .filter(|&&key| key == button)
                 .take(MAX as usize)
-                .count() as u8,
-            Button::Mouse(button) => self.delta
+                .count() as Self::Output,
+            Button::Mouse(button) => input.delta
                 .mouse_hit
                 .iter()
                 .filter(|&&key| key == button)
                 .take(MAX as usize)
-                .count() as u8,
-        };
-        ButtonInfo {
-            pressed,
-            hits,
+                .count() as Self::Output,
         }
     }
+}
 
-    pub fn key_axis(
-        &self,
-        axis: axis::Key,
-    ) -> KeyAxisInfo {
-        let state = {
-            let is_pos = self.state.keys_pressed.contains(&axis.pos);
-            let is_neg = self.state.keys_pressed.contains(&axis.neg);
-            if is_pos && !is_neg {
-                Some(1)
-            } else if is_neg && !is_pos {
-                Some(-1)
-            } else {
-                None
-            }
-        };
-        let timed_state = state.map(|v| v as f32 * self.delta_time());
-        let mut hits = KeyAxisHits::default();
-        hits.pos = self.delta
+impl HitCount for axis::Key {
+    type Output = (u8, u8);
+
+    fn hit_count(&self, input: &Input) -> Self::Output {
+        use std::u8::MAX;
+        let pos = input.delta
             .keys_hit
             .iter()
-            .filter(|&&k| k == axis.pos)
+            .filter(|&&k| k == self.pos)
+            .take(MAX as usize)
             .count() as u8;
-        hits.neg = self.delta
+        let neg = input.delta
             .keys_hit
             .iter()
-            .filter(|&&k| k == axis.neg)
+            .filter(|&&k| k == self.neg)
+            .take(MAX as usize)
             .count() as u8;
-        let delta_hits = if hits.pos + hits.neg == 0 {
+        (pos, neg)
+    }
+}
+
+pub trait Delta {
+    type Output;
+
+    fn delta(&self, input: &Input) -> Self::Output;
+    fn timed_delta(&self, input: &Input) -> Option<TimerDuration>;
```

Comment:
I think `timed` is good enough.

Reply:
Yep, maybe. Also we can reduce code a bit by providing default impl for this method

Reply:
Are these methods independent? If so, perhaps we should split them into `Delta` and `TimedDelta`.